src/core_plugins/input_control_vis/public/control/control.js

```diff
@@ -7,17 +7,74 @@ Choose a different field or index documents that contain values for this field.`
 }
 
 export class Control {
-  constructor(controlParams, filterManager) {
+  constructor(controlParams, filterManager, kbnApi, useTimeFilter) {
     this.id = controlParams.id;
     this.options = controlParams.options;
     this.type = controlParams.type;
     this.label = controlParams.label ? controlParams.label : controlParams.fieldName;
+    this.useTimeFilter = useTimeFilter;
     this.filterManager = filterManager;
-    this.enable = true;
+    this.kbnApi = kbnApi;
+
     // restore state from kibana filter context
     this.reset();
+    // disable until initialized
+    this.disable('Control has not been initialized');
+  }
+
+  async fetch() {
+    throw new Error('fetch method not defined');
+  }
+
+  /**
+   *
+   * @param ancestors {array of Controls}
+   */
+  setAncestors(ancestors) {
+    this.ancestors = ancestors;
+  }
+
+  hasAncestors() {
+    return this.ancestors && this.ancestors.length > 0;
   }
 
+  hasUnsetAncestor() {
+    return this.ancestors.reduce((accumulator, ancestor) => {
+      return accumulator || !ancestor.hasValue();
+    }, false);
+  }
+
+  getAncestorSignature() {
+    return this.ancestors.map(ancestor => {
+      return ancestor.value;
+    }).join(',');
+  }
+
+  getAncestorSearchSource() {
+    let previousAncestorSearchSource;
```

Comment:
Why are we making all of the search sources inherit from their parent/grandparents/etc.? Don't we just want to be using the parent's value as the filter? That doesn't seem like it'd require walking up the parent hierarchy more than your direct parent.

Reply:
You need the entire ancestry chain to ensure nested controls only contain options for the entire chain of selected values. 

For example control1 = state, control2 = city, control3 = street name. There is a Rome Georgia and a Rome Ohio. If you only filtered by the parent control then your street name control would get results for all cities with the name Rome. But if you filter by the entire ancestry chain then the street name control would only get results where the State is Georgia and the City is Rome.

Reply:
Gotcha, that makes sense why we'd need to add the filters for all ancestors, but I'm still not following why we need the search source inheritance.